keras/utils/conv_utils.py

```diff
@@ -85,7 +85,20 @@ def normalize_tuple(value, n, name):
         error_msg += (f'including element {single_value} of '
                       f'type {type(single_value)}')
         raise ValueError(error_msg)
-    return value_tuple
+
+  if allow_zero:
+    qualified_values = list(filter(lambda x: x >= 0, value_tuple))
+    req_msg = '>=0'
+  else:
+    qualified_values = list(filter(lambda x: x > 0, value_tuple))
+    req_msg = '>0'
+
+  if len(qualified_values) < len(value_tuple):
+    error_msg += (f' including {qualified_values}'
```

Comment:
Shouldn't the error message show the dis-qualified value, rather than the qualified ones?